tools/src/launcher/Main.hs

```diff
@@ -128,12 +129,20 @@ data NodeData = NodeData
     , ndArgs    :: ![Text]
     , ndLogPath :: Maybe FilePath }
 
--- | Updater path, args, windows runner path, archive path
-data UpdaterData = UpdaterData
-    { udPath        :: !FilePath
-    , udArgs        :: ![Text]
-    , udWindowsPath :: Maybe FilePath
-    , udArchivePath :: Maybe FilePath
+data UpdaterData = WindowsUpdaterData WinUD | OSXUpdaterData OSXUD | UnsupportedOSUpdateData UnsupOSUD
+
+data UnsupOSUD = UnsupOSUD
+
+data WinUD = WinUD 
+    { wudExePath :: !FilePath -- Installer.exe
+    , wudArgs    :: ![Text]
+    , wudBatPath :: !FilePath -- Installer.bat
+    }
+
+data OSXUD = OSXUD
```

Comment:
We tend to avoid so long abbreviations, because it's hard to distinguish parts of it. You can call it `OsxUD` or maybe `MacUD` or `MacOsUD`.